src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -0,0 +1,502 @@
+     */
+    private List<Geometry> convertToJtsGeometryAndSlice(final Line line)
+    {
+        // Create the JTS Geometry from Line
+        final Geometry geometry;
+        if (line.isClosed())
+        {
+            // An Area
+            geometry = JTS_POLYGON_CONVERTER.convert(new Polygon(line));
+        }
+        else
+        {
+            // A Line
+            geometry = JTS_POLYLINE_CONVERTER.convert(line.asPolyLine());
+        }
+
+        // Slice the JTS Geometry
+        try
+        {
+            return this.countryBoundaryMap.slice(line.getIdentifier(), geometry);
+        }
+        catch (final TopologyException e)
+        {
+            logger.error("Topology Exception when slicing {}", line.getIdentifier(), e);
+            return Collections.emptyList();
+        }
+    }
+
+    /**
+     * Assigns {@link ISOCountryTag} and {@link SyntheticNearestNeighborCountryCodeTag} values for a
+     * given {@link Geometry}, which is a {@link Line} since we don't have other non-Point
+     * geometries in the raw Atlas.
+     *
+     * @param geometry
+     *            The {@link Geoemtry} to create the tags for
+     * @return the resulting tags
+     */
+    private Map<String, String> createLineTags(final Geometry geometry)
+    {
+        final Map<String, String> tags = new HashMap<>();
+        final String countryCode = CountryBoundaryMap.getGeometryProperty(geometry,
+                ISOCountryTag.KEY);
+        tags.put(ISOCountryTag.KEY, countryCode);
+        final String usingNearestNeighbor = CountryBoundaryMap.getGeometryProperty(geometry,
+                SyntheticNearestNeighborCountryCodeTag.KEY);
+        if (usingNearestNeighbor != null)
+        {
+            tags.put(SyntheticNearestNeighborCountryCodeTag.KEY, usingNearestNeighbor);
+        }
+        return tags;
+    }
+
+    /**
+     * Creates a new {@link TemporaryPoint} at the given {@link Coordinate}. We can assume that any
+     * new Points will be created at the country boundaries, so we can add the
+     * {@link SyntheticBoundaryNodeTag} tag.
+     *
+     * @param coordinate
+     *            The {@link Coordinate} of the new point
+     * @param pointIdentifierFactory
+     *            The {@link CountrySlicingIdentifierFactory} to calculate new point identifier
+     * @param pointTags
+     *            The tags for this new point
+     * @return an {@link Optional} containing the {@link TemporaryPoint}
+     */
+    private Optional<TemporaryPoint> createNewPoint(final Coordinate coordinate,
+            final CountrySlicingIdentifierFactory pointIdentifierFactory,
+            final Map<String, String> pointTags)
+    {
+        if (!pointIdentifierFactory.hasMore())
+        {
+            logger.error(
+                    "Country Slicing exceeded maximum number of supported new points at Coordinate {}",
+                    AbstractIdentifierFactory.IDENTIFIER_SCALE, coordinate);
+            return Optional.empty();
```

Comment:
Good callout! Thought about this for a while and went back and forth on what to do. If we throw an Exception, we'll stop country-slicing for this RawAtlas and essentially cease Atlas creation until either the code is changed (refactor identifiers, filter types of ways) or OSM data changes (we don't want this). Two feature types that are prone to this are boundaries and coastlines. 

I'd like to avoid this and handle it more gracefully. Some ideas: 1. Throw a specific type of error, log and drop this way from the raw Atlas. 2. Cease slicing once we reach the maximum identifier and extend the last segment to be a single long segment until the end. 

Option 1 seems cleaner and Option 2 is error-prone and confusing, since you will have a (potentially long) segment spanning 2 (or more) countries with a single country code. 

Thoughts?

Reply:
I agree, option 1 is better! Thanks!